src/components/TimeCounter.tsx

```diff
@@ -1,79 +1,67 @@
-import { StackNavigationProp } from '@react-navigation/stack';
-import dayjs from 'dayjs';
-import React, { useState, useEffect } from 'react';
-import { StyleSheet, Text, View, KeyboardAvoidingView } from 'react-native';
-import { Overlay } from 'react-native-elements';
+import React, { Component } from 'react';
+import { View, StyleSheet } from 'react-native';
 
-import { MainCardStackNavigatorParams, Route } from 'app/consts';
-import { useInterval } from 'app/helpers/useInterval';
-import { typography } from 'app/styles';
+import { palette, fonts } from 'app/styles';
 
-import { Button } from './Button';
-
-const i18n = require('../../loc');
+import { Text } from './Text';
 
 interface Props {
-  timestamp: number;
-  isVisible: boolean;
-  navigation: StackNavigationProp<MainCardStackNavigatorParams, Route.CurrentPin>;
-  onCountFinish?: () => void;
+  value: number;
 }
 
-export const TimeCounter = ({ timestamp, onCountFinish, isVisible, navigation }: Props) => {
-  const currentTimestamp = dayjs().unix();
-  const secondsToCount = (timestamp - currentTimestamp).toFixed(0);
-  const [seconds, setSeconds] = useState(parseInt(secondsToCount));
-
-  const isTimeUp = (): boolean => seconds === 0;
-
-  useInterval(
-    () => {
-      setSeconds(seconds - 1);
-    },
-    seconds > 0 ? 1000 : null,
-  );
-
-  useEffect(() => {
-    if (isTimeUp()) {
-      onFinish();
-    }
-  });
-
-  const onFinish = () => {
-    if (onCountFinish) {
-      onCountFinish();
-    }
-  };
+export class TimeCounter extends Component<Props> {
+  render() {
+    const { value } = this.props;
+    const minutes = Math.floor(value / 60);
```

Comment:
please don't manipulate time on your own. We have dayjs for this. You can write it like this 
![Zrzut ekranu 2020-08-11 o 08 40 14](https://user-images.githubusercontent.com/67273035/89865890-d6a2b000-dbae-11ea-9cae-78f2a4ec9e81.png)
Create a function in utils/date.ts for this. Something like const `secondsToFormat = (seconds, format) => ...`